src/main/java/de/tum/p2p/api/netty/forwarding/OnionCoverMessage.java

```diff
@@ -0,0 +1,31 @@
+package de.tum.p2p.api.netty.forwarding;
+
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+import lombok.NonNull;
+
+import de.tum.p2p.proto.message.MessageType;
+import java.nio.ByteBuffer;
+
+@ToString
+@EqualsAndHashCode(callSuper = true)
+public class OnionCoverMessage extends OnionApiMessage {
+
+    @NonNull
+    private final int coverSize;
+
+    public OnionCoverMessage(int coverSize) {
+        super(MessageType.ONION_COVER);
+        this.coverSize = coverSize;
+    }
+
+    @Override
+    protected ByteBuffer writeMessage(ByteBuffer typedMessageBuffer) {
+        return typedMessageBuffer; // Onion Cover message carries no payload
```

Comment:
If you dont expect this to be used/useful throw new UnsupportedOperationException() here